src/VisualStudio/TestUtilities/Remoting/RemotingWorkspace.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.Shell;
+using RoslynProject = Microsoft.CodeAnalysis.Project;
+using RoslynWorkspace = Microsoft.CodeAnalysis.Workspace;
+
+namespace Roslyn.VisualStudio.Test.Utilities.Remoting
+{
+    internal class RemotingWorkspace
+    {
+        public static void AddReference(string projectName, string referenceAssemblyPath)
+        {
+            var metadataReference = MetadataReference.CreateFromFile(referenceAssemblyPath);
+            RoslynWorkspace workspace = GetWorkspace();
+            RoslynProject newProject = GetProject(workspace, projectName).AddMetadataReference(metadataReference);
+            workspace.TryApplyChanges(newProject.Solution);
+        }
+
+        public static void OpenDocument(string projectName, string documentName)
+        {
+            RoslynWorkspace workspace = GetWorkspace();
+            RoslynProject project = GetProject(workspace, projectName);
+            DocumentId documentId = project.Documents.FirstOrDefault(doc => doc.Name.Equals(documentName)).Id;
+
+            if (documentId != null)
+            {
+                RemotingHelper.ExecuteOnUIThread(() => workspace.OpenDocument(documentId, activate: true));
+            }
+        }
+
+        private static RoslynWorkspace GetWorkspace()
```

Comment:
For most items (especially things such as the `Workspace` or `InteractiveWindow`, which represent a service), the convention should be to create a wrapper class around the object and then have a static method in `RemotingHelper` which returns an instance of said wrapper.

`GetActiveTextViewHost` is somewhat special because it is not a service and two sequential calls can potentially return different interfaces. It should (therefore) not be marshalled across the process boundary as it can lead to some hard-to-track down bugs (and can mess with the object lifetime in ways VS doesn't expect).